test/parallel/test-https-agent.js

```diff
@@ -57,7 +61,8 @@ server.listen(0, function() {
         }, function(res) {
           res.resume();
           assert.strictEqual(res.statusCode, 200);
-          if (++responses === N * M) server.close();
+          ++responses;
```

Comment:
By adding the countdown you can eliminate the responses counter and the process.on('exit') check